pilot/pkg/kube/inject/webhook_test.go

```diff
@@ -307,6 +320,402 @@ func TestInject(t *testing.T) {
+			wantFile:  "frontend.yaml.injected",
+		},
+		{
+			inputFile: "hello-multi.yaml",
+			wantFile:  "hello-multi.yaml.injected",
+		},
+		{
+			inputFile: "multi-init.yaml",
+			wantFile:  "multi-init.yaml.injected",
+		},
+		{
+			inputFile: "statefulset.yaml",
+			wantFile:  "statefulset.yaml.injected",
+		},
+		{
+			inputFile: "daemonset.yaml",
+			wantFile:  "daemonset.yaml.injected",
+		},
+		{
+			inputFile: "job.yaml",
+			wantFile:  "job.yaml.injected",
+		},
+		{
+			inputFile: "replicaset.yaml",
+			wantFile:  "replicaset.yaml.injected",
+		},
+		{
+			inputFile: "replicationcontroller.yaml",
+			wantFile:  "replicationcontroller.yaml.injected",
+		},
+		{
+			inputFile: "list.yaml",
+			wantFile:  "list.yaml.injected",
+		},
+		{
+			inputFile: "list-frontend.yaml",
+			wantFile:  "list-frontend.yaml.injected",
+		},
+		{
+			inputFile: "deploymentconfig.yaml",
+			wantFile:  "deploymentconfig.yaml.injected",
+		},
+		{
+			inputFile: "deploymentconfig-multi.yaml",
+			wantFile:  "deploymentconfig-multi.yaml.injected",
+		},
+		{
+			// Verifies that annotation values are applied properly. This also tests that annotation values
+			// override params when specified.
+			inputFile: "traffic-annotations.yaml",
+			wantFile:  "traffic-annotations.yaml.injected",
+		},
+		{
+			// Verifies that the wildcard character "*" behaves properly when used in annotations.
+			inputFile: "traffic-annotations-wildcards.yaml",
+			wantFile:  "traffic-annotations-wildcards.yaml.injected",
+		},
+		{
+			// Verifies that the wildcard character "*" behaves properly when used in annotations.
+			inputFile: "traffic-annotations-empty-includes.yaml",
+			wantFile:  "traffic-annotations-empty-includes.yaml.injected",
+		},
+	}
+
+	for _, c := range cases {
+		input := filepath.Join("testdata", c.inputFile)
+		want := filepath.Join("testdata", c.wantFile)
+		t.Run(c.inputFile, func(t *testing.T) {
+			// Split multi-part yaml documents. Input and output will have the same number of parts.
+			inputYAMLs := splitYamlDoc(input, t)
+			wantYAMLs := splitYamlDoc(want, t)
+
+			for i := 0; i < len(inputYAMLs); i++ {
+				t.Run(fmt.Sprintf("yamlPart[%d]", i), func(t *testing.T) {
+					// Convert the input YAML to a deployment.
+					inputYAML := inputYAMLs[i]
+					inputJSON := yamlToJSON(inputYAML, t)
+					inputDeployment := jsonToDeployment(inputJSON, t)
+
+					// Convert the wanted YAML to a deployment.
+					wantYAML := wantYAMLs[i]
+					wantJSON := yamlToJSON(wantYAML, t)
+					wantDeployment := jsonToDeployment(wantJSON, t)
+
+					// Generate the patch.  At runtime, the webhook would actually generate the patch against the
+					// pod configuration. But since our input files are deployments, rather than actual pod instances,
+					// we have to apply the patch to the template portion of the deployment only.
+					templateJSON := toJSON(inputDeployment.Spec.Template, t)
+					wh := createTestWebhook(configMapFile.Name(), t)
```

Comment:
Can the webhook instance be shared across test cases in this test?

Reply:
yes, done.